src/nginx/module.cc

```diff
@@ -42,45 +42,70 @@
 #include "src/nginx/util.h"
 #include "src/nginx/version.h"
 
-using ::google::protobuf::util::error::Code;
 using ::google::api_manager::utils::Status;
+using ::google::protobuf::util::error::Code;
 
 namespace google {
 namespace api_manager {
 namespace nginx {
 
 namespace {
 
-// Updates r->uri and r->unparsed_uri, based on api basepath math.
-// Then returns ApiManager::ApiBasepathRewriteAction.
-// If the return value is ApiManager::ApiBasepathRewriteAction::REJECT,
-// then nginx should response status 404 NotFound
-ApiManager::ApiBasepathRewriteAction ngx_esp_rewrite_uri(
-    ngx_http_request_t *r, ngx_esp_loc_conf_t *lc) {
-  std::string unparsed_uri;
-  ApiManager::ApiBasepathRewriteAction unparsed_uri_action =
-      lc->esp->ReWriteURL(ngx_str_to_std(r->unparsed_uri), &unparsed_uri);
+// Internal debugging header
+std::string kXEndpointsDebugUrlRewrite = "x-endpoints-debug-url-rewrite";
```

Comment:
use ngx_str_t may be easier to use than std::string

Reply:
Done